docs/rag_architecture.md

```diff
@@ -1,7 +1,67 @@
 # wise-agents RAG architecture
 
-## GraphRAG: a description
+Wise Agents provides agent implementations that can be used for both standard retrieval augmented
+generation (RAG) and also for RAG with knowledge graphs (Graph RAG).
 
-## Why a GraphRAG?
+## What is RAG?
 
-## Our architecture and implementations
\ No newline at end of file
+Retrieval augmented generation (RAG) provides the ability to enhance the knowledge of a large language
+model (LLM) by incorporating additional information during response generation. This is very important
+because although LLMs are trained with extremely large quantities of data, this data is static, has
+a cut-off date, and likely lacks domain-specific information. RAG provides the ability to incorporate
+additional information obtained from knowledge sources to ground LLM responses using this information,
+resulting in better quality responses.
+
+### How does RAG work?
+
+To be able to make use of RAG, the very first step is ingestion. Here, knowledge sources are processed,
+split into chunks of text, each chunk of text is converted to a vector representation (also known as
+an embedding), and then stored in a vector database.
+
+After this initial step, upon receiving a query, the goal is to retrieve the information that is most
+relevant to the query from the vector database. This is done by converting the query to a vector
+representation (i.e., embedding) and then querying the vector database to retrieve the chunks that
+are most similar to the query.
+
+The retrieved information is then included in a prompt to the LLM along with the original query. This
+helps the LLM to generate a response that is grounded in the retrieved information.
+
+## Our architecture and implementations
+
+TODO
+
+## What is Graph RAG?
+
+Graph RAG is a more structured approach to RAG. In standard RAG, as described above, the knowledge
+sources are simply split into chunks of text. However, there are many sources of information that
+actually have a lot of structure to them where this structure would be lost if just convert the knowledge
+source to text as is and split into chunks. For example, think about a bug report that has many different
+fields like "Description", "Steps to Reproduce", "Relates To", "Resolution", etc. If we just take the text
```

Comment:
Maybe could be worth pointing out that i we find a match in the Description, we could feed the contents of the Resolution back to the LLM.  

Not a must, just a nice-to-have :-)

Reply:
Thanks! I've updated the "How does Graph RAG work?" section to mention this.